File: src/problems_all/191_number_of_1_bits/191_number_of_1_bits_v1.cpp
/*
https://leetcode.com/problems/number-of-1-bits/description/

Runtime: 0 ms
Memory: 6.1 MB
*/

//#include <bitset>
//#include <string>
//#include <iostream>
#include <bits/stdc++.h>


class Solution
{
public:
    int hammingWeight(uint32_t n)
    {
        std::string bin = std::bitset<64>(n).to_string();
        std::string::difference_type ones = std::count(bin.begin(), bin.end(), '1');

        return ones;
    }
};



int main()
{
    uint32_t n = 2147483647;
    Solution* solution = new Solution();
    int ones = solution->hammingWeight(n);
    std::cout << ones << std::endl;

    return 0;
}

File: src/problems_all/191_number_of_1_bits/191_number_of_1_bits_v1.py
"""
https://leetcode.com/problems/number-of-1-bits/description/

Runtime: 41 ms
Memory: 13.9 MB
"""


class Solution:
    def hammingWeight(self, n: int) -> int:
        bstr = bin(n)[2:]
        total = 0
        for i in bstr:
            if int(i) == 1:
                total += 1

        return total


def main():
    n = 123
    weight = Solution().hammingWeight(n)
    print(weight)


if __name__ == '__main__':
    main()
